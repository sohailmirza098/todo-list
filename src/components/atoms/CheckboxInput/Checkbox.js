// import React from 'react'

// export default function Checkbox(props) {
//     return (
//         <input type="checkbox" onChange={ () => props.markComplete(props.id) }/>
//     )
// }


import React from 'react'


const Checkbox = (props, { children }) => (
    <input type="checkbox" onChange={ () => props.markComplete(props.id) }/>
)

export default Checkbox; 