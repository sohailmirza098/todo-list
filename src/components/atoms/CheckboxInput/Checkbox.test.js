import React from "react";
import { create } from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Checkbox from "./Checkbox";

describe("Checkbox testing", () => {

  test("Matches the snapshot", () => {
    const input = create(<Checkbox />);
    expect(input.toJSON()).toMatchSnapshot();
  });

  test("Checkbox is present", () => {
    const {getByRole}=  render(<Checkbox />);
    const checkbox = getByRole("checkbox");
    expect(checkbox).toBeInTheDocument();
  });

  test("Checkbox not marked", () => {
    const {getByRole}=  render(<Checkbox />);
    const checkbox = getByRole("checkbox");
    expect(checkbox.checked).toEqual(false);
  });

  test("Checkbox marked",()=>{
    const {getByRole}=  render(<Checkbox />);
    const checkbox = getByRole("checkbox");
    fireEvent.click(checkbox)
    expect(checkbox.checked).toEqual(true)
  });

  test("Checkbox onclick",()=>{
    const mockClickMethod = jest.fn();
    const {getByRole}=  render(<Checkbox onClick={mockClickMethod()}/>);
    fireEvent.click(getByRole("checkbox"));
    expect(mockClickMethod).toHaveBeenCalledTimes(1);
  });

  
});