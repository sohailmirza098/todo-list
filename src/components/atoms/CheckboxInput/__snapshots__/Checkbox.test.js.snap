exports[`Checkbox testing Matches the snapshot 1`] = `
<input
  onChange={[Function]}
  type="checkbox" />
`;
